### PR TITLE
Use spdlog for logging

### PR DESCRIPTION
**Pull request by morxa**
Instead of printing to `cout`, use spdlog for logging. This allows us to later log to a file or set a certain logging level (e.g., for tracing).

### CODE REVIEW

File: src/synchronous_product/include/synchronous_product/search.h [hunk 1]
```diff
@@ -127,7 +129,7 @@ class TreeSearch
 		}
 		Node *current = queue_.front();
 		queue_.pop();
-		std::cout << "Processing " << current << ": " << *current;
+		// SPDLOG_TRACE("Processing {}: {}", current, *current);
```
Comment (modass) [inline src/synchronous_product/include/synchronous_product/search.h:132]: Why is this commented?

**Review by modass — APPROVED** [review]
Looks straight forward

File: src/synchronous_product/include/synchronous_product/search.h (hunk 1 above)
Comment (morxa) [reply src/synchronous_product/include/synchronous_product/search.h]: Good catch, fixed!

**Comment by morxa** [comment]
This needed some additional changes after rebasing, in particular c6c6ef690be3d821c29183537ea8d1f105dbe643. Please have another look :)

**Review by modass — COMMENTED** [review]
This feels like we may separate even further: 
- Create a header "types.h" or "typedefs.h" or "common.h" (I do not like the last option, but I have seen it) and put all using-declarations and includes there.
- Create a header "IO.h" or similar which defines all streaming operators.

What do you think?

**Review by modass — APPROVED** [review]
As discussed

**Pull request merged**